src/colorizer/viewport/tracks/TrackPath2D.ts

```diff
@@ -233,8 +239,21 @@ export default class TrackPath2D {
       }
     }
 
-    (this.line.material as SubrangeLineMaterial).minInstance = startingInstance;
-    (this.bgLine.material as SubrangeLineMaterial).minInstance = startingInstance;
+    if (
+      !(this.line.material instanceof SubrangeLineMaterial) ||
+      !(this.bgLine.material instanceof SubrangeLineMaterial)
+    ) {
+      return;
+    }
+
+    // Update color ramp related logic
+    const pastSteps = this.params.showAllTrackPathPastSteps ? track.duration() : this.params.trackPathPastSteps;
+    const futureSteps = this.params.showAllTrackPathFutureSteps ? track.duration() : this.params.trackPathFutureSteps;
+    this.line.material.colorRampVertexOffset = trackStepIdx;
+    this.line.material.colorRampVertexScale = Math.max(pastSteps, futureSteps);
```

Comment:
Should this be x2 here? or in the shader?

- [ ] Explicitly document what the value of the scale is